core/src/main/java/org/apache/carbondata/core/datastore/row/CarbonRow.java

```diff
@@ -31,6 +34,9 @@
 
   private short rangeId;
 
+  /* key is complex column name, val is it's flat byte array */
+  private Map<String, List<ArrayList<byte[]>>> complexFlatByteArrayMap;
```

Comment:
Why it is a separate map, It should be part of `data` array

Reply:
This is an extracted data from `data` of `CarbonRow`.
previously in `TablePage.java`, List<ArrayList<byte[]>> encodedComplexColumnar was extracted  from `data` and using it, it was never modied actual `data`. So, I have used the same logic.
Also modifying data[] with flat complex data may lead to many code changes where data[] is used.

Reply:
yes, handled